app/pages/learn/01_tutorial/04_mastering-the-api/02_invoke/00_methodhandle.md

```diff
@@ -0,0 +1,531 @@
+These include method handles that allow array manipulations. Creating these method handles doesn't require access checking, so the lookup object is not necessary.
+
+Let's create an array of Strings containing 5 elements using `arrayConstructor`:
+
+```java
+MethodHandle arrayConstructor = MethodHandles.arrayConstructor(String[].class);
+String[] arr = (String[]) arrayConstructor.invoke(5);
+```
+
+To modify a single element, we can use `arrayElementSetter`, to which we provide the reference to the target array, the index of an element, and the new value:
+
+```java
+MethodHandle elementSetter = MethodHandles.arrayElementSetter(String[].class);
+elementSetter.invoke(arr, 4, "test");
+```
+
+To read the value of a single element, we should use `arrayElementGetter` method handle, to which we provide the reference to an array and the element index:
+
+```java
+MethodHandle elementGetter = MethodHandles.arrayElementGetter(String[].class);
+String element = (String) elementGetter.invoke(arr, 4); // element == "test"
+```
+
+We could also use the method handle provided by `arrayLength` to get the array length as integer:
+
+```java
+MethodHandle arrayLength = MethodHandles.arrayLength(String[].class);
+int length = (int) arrayLength.invoke(arr); // length == 5
+```
+
+<a id="vsreflection">&nbsp;</a>
+## Method handles vs Reflection
+The Method Handles mechanism was not designed for wide usage. Originally, it was meant as a tool to assist compiler and language runtime developers. 
+It has a narrower range of capabilities than the Reflection API. 
+For example, the Reflection API allows listing the class members and inspecting their properties, which cannot be done using method handles.
+
+There are also certain differences when it comes to access checking and security considerations. 
+The Core Reflection API performs access checking against every caller, on every call, while for method handles, access is checked only during construction. 
+This makes invocation through method handles faster than through reflection. 
+However, certain precautions have to be taken so the method handle is not passed to the code where it shouldn't be accessible.
+
+Method handles and reflection serve different purposes, so it's important to pick the right tool based on your needs.
+
+<a id="unreflect">&nbsp;</a>
+## Conversion between Reflection API and method handles
+The `Lookup` object can be used to convert Reflection API objects to behaviorally equivalent method handles, which provide more direct and efficient access to the underlying class members.
+
+To create a method handle pointing to a given `Method` (given that the lookup class has permission to do so), we can use `unreflect`.
+
+Let's say we have a `test` method in our `Example` class which accepts a `String` argument and returns a `String`. 
+Using the Reflection API, we can obtain a `Method` object:
+
+```java
+Class clazz = Class.forName("Example");
```

Comment:
Is there any reason you used `Class.forName` instead of just `Example.class` here? Doing it that way just makes it require the class to be in the default package.